xlators/features/shard/src/shard.c

```diff
@@ -49,15 +51,19 @@ shard_make_block_bname(int block_num, uuid_t gfid, char *buf, size_t len)
     snprintf(buf, len, "%s.%d", gfid_str, block_num);
 }
 
-void
-shard_make_block_abspath(int block_num, uuid_t gfid, char *filepath, size_t len)
+static int
+shard_make_base_path(char *path, uuid_t gfid)
 {
-    char gfid_str[GF_UUID_BUF_SIZE] = {
-        0,
-    };
+    strcpy(path, "/" GF_SHARD_DIR "/");
+    uuid_utoa_r(gfid, path + sizeof(GF_SHARD_DIR) + 1);
+    return (sizeof(GF_SHARD_DIR) + GF_UUID_BUF_SIZE);
```

Comment:
what about the "/" before and after GF_SHARD_DIR?

Reply:
after the strcpy, the path would contain "/.shard/"